### PR TITLE
fix CTE and UNION case in #481

### PR DESCRIPTION
**Pull request by maoxingda**
making progress in #481

### CODE REVIEW

File: sqllineage/core/parser/sqlfluff/extractors/select.py [hunk 1]
```diff
@@ -150,10 +160,12 @@ def _handle_set(self, segment: BaseSegment) -> None:
                     subquery, alias = sq
                     table_identifier = find_table_identifier(subquery)
                     if table_identifier:
-                        read_sq = SqlFluffTable.of(table_identifier, alias)
                         for seg in list_child_segments(subquery):
+                            for set_subquery in self.list_subquery(seg):
+                                self.subqueries_within_set.append(set_subquery)
+
+                            self._handle_table(seg, holder)
                             self._handle_column(seg)
```
Comment (reata) [inline sqllineage/core/parser/sqlfluff/extractors/select.py:168]: Let's remove the `list_subqueries` call. We can get children segment of type `select_statement` and `bracketed` from set_expression. And then list_child_segments from `select_statement` and `bracketed` and feed it to `_handle_table` and `_handle_column`.  We don't need subqueries_within_set that way.

File: tests/sql/column/test_column_cross_stmts_with_similar_subquery_alias.py
Comment (reata) [inline tests/sql/column/test_column_cross_stmts_with_similar_subquery_alias.py:1]: This is a separate issue. Create another PR for this.

**Review by reata — CHANGES_REQUESTED** [review]
Thanks for the PR. Here's the review summary:

1) We don't really need to modify holder implementation. Just try re-use `_handle_table` and `_handle_column` call in `_handle_set` method.
2) Focus on the test case with CTE + UNION on this PR. Only Keep the test case tests/sql/column/test_column_subquery_within_set.py.

File: sqllineage/core/parser/sqlfluff/extractors/select.py (hunk 1 above)
Comment (maoxingda) [reply sqllineage/core/parser/sqlfluff/extractors/select.py:168]: I don't quite understand. Can you help me by directly modifying my code?

File: tests/sql/column/test_column_cross_stmts_with_similar_subquery_alias.py
Comment (maoxingda) [reply tests/sql/column/test_column_cross_stmts_with_similar_subquery_alias.py:1]: Okay, I will create a separate pull request.

**Comment by reata** [comment]
Code modified and merged. This PR is narrowed to handle CTE+UNION case only. Will handle other cases in original issue in separate PR.

**Pull request merged**